packages/frontend/editor-ui/src/init.ts

```diff
@@ -152,3 +162,30 @@ export async function initializeAuthenticatedFeatures(
 
 	authenticatedFeaturesInitialized = true;
 }
+
+function registerAuthenticationHooks() {
+	const rootStore = useRootStore();
+	const usersStore = useUsersStore();
+	const cloudPlanStore = useCloudPlanStore();
+	const postHogStore = usePostHog();
+	const uiStore = useUIStore();
+	const npsSurveyStore = useNpsSurveyStore();
+	const telemetry = useTelemetry();
+	const RBACStore = useRBACStore();
+
+	usersStore.registerLoginHook((user) => {
+		RBACStore.setGlobalScopes(user.globalScopes ?? []);
+		telemetry.identify(rootStore.instanceId, user.id);
+		postHogStore.init(user.featureFlags);
+		npsSurveyStore.setupNpsSurveyOnLogin(user.id, user.settings);
+	});
+
+	usersStore.registerLogoutHook(() => {
+		RBACStore.setGlobalScopes([]);
+		telemetry.reset();
+		cloudPlanStore.reset();
+		postHogStore.reset();
+		uiStore.clearBannerStack();
+		npsSurveyStore.resetNpsSurveyOnLogOut();
+	});
+}
```

Comment:
1. Should PostHog be part of the existing `telemetry`?
2. Shouldn't the logout hooks run in reverse? So RBAC is the last thing on the way out?

Reply:
Updated the order. Posthog is using a separate store entirely. We'll probably merge them at some point.